src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -586,6 +602,69 @@ export class ResultsViewPageStore extends AnalysisStore
         | ModifyQueryParams
         | undefined = undefined;
 
+    @action.bound
+    public setUserSelectedClinicalTrackColor(
+        label: string,
+        value: string,
+        color: RGBAColor | undefined
+    ) {
+        // if color is undefined, delete color from userSelectedClinicalAttributeColors if exists
+        // else, set the color in userSelectedClinicalAttributeColors
+        if (
+            !color &&
+            this._userSelectedClinicalTracksColors[label] &&
+            this._userSelectedClinicalTracksColors[label][value]
+        ) {
+            delete this._userSelectedClinicalTracksColors[label][value];
+        } else if (color) {
+            if (!this._userSelectedClinicalTracksColors[label]) {
+                this._userSelectedClinicalTracksColors[label] = {};
+            }
+            this._userSelectedClinicalTracksColors[label][value] = color;
+        }
+    }
+
+    @computed get defaultClinicalTracksColors() {
+        return this._defaultClinicalTracksColors;
+    }
+
+    @computed get userSelectedClinicalTracksColors() {
+        return this._userSelectedClinicalTracksColors;
+    }
+
+    @action.bound
+    public setDefaultClinicalTrackColors(
```

Comment:
we shouldn't ever need to set the defaults.  when the consumer of the track colors (the oncoprint rendring logic) requests a color, we look to see if there is a custom color for that track, and if not, we use a default.   since that default is not static, we may need to "cache" the colors so that they are only generated once during a load of the page.